digdag-standards/src/main/java/io/digdag/standards/operator/jdbc/AbstractJdbcOperator.java

```diff
@@ -180,7 +180,11 @@ public TaskResult runTask()
                     logger.warn("Error during cleaning up status table. Ignoring.", ex);
                 }
 
-                return TaskResult.defaultBuilder(request).build();
+                // It's possible that the query already finished successfully, but this task don't know that
+                // if the previous connection with database was disconnected during waiting result.
+                // So setting `completedAt` in state is needed even if the task was already finished.
```

Comment:
I don't know why this is needed...as I commented at https://github.com/treasure-data/digdag/pull/179#discussion_r72129236.
If here calls REST API to set only state params, this is good. Because setting state params only probably less likely fails compared to setting the state to finished, and its overhead is small. The idea to add TaskContext.setStateParams is as commented at: https://github.com/treasure-data/digdag/pull/177#discussion_r71963995
But here doesn't use it. Making the task entry on the queue finished and retrying it again involves large overhead.
Without this call, this agent immediately calls back to digdag-server to mark the task finished. Once the task is marked as finished successfully, the task won't be re-executed. The risk of failure of the callback is same with the risk of failure of `ofNextPolling` callback.